### PR TITLE
(Refactor)[Nereids] Combine operator and plan

### PR DESCRIPTION
**Pull request by 924060929**
# Proposed changes

Issue Number: no issue

## Problem Summary:

in #9755, we split plan into plan & operator, but in subsequent development, we found the rule became complex and counter intuition: 
1. we must create an operator instance, then wrap a plan by the operator type.
2. relational algebra(operator) not contains children 

e.g.
```java
logicalProject().then(project -> {
    List<NamedExpression> boundSlots =
        bind(project.operator.getProjects(), project.children(), project);
    LogicalProject op = new LogicalProject(flatBoundStar(boundSlots));
    // wrap a plan
    return new LogicalUnaryPlan(op, project.child());
})
```

after combine operator and plan, the code become to:
```java
logicalProject().then(project -> {
    List<NamedExpression> boundSlots =
        bind(project.getProjects(), project.children(), project);
    return new LogicalProject(flatBoundStar(boundSlots), project.child());
})
```

Originally, we thought it would be convenient for `Memo.copyIn()` after split plan & operator, because Memo don't known how to re-new the plan(assembling child plan in the children groups) by the plan type. So plan must provide the `withChildren()` abstract method to assembling children. The less plan type, the lower code cost we have(logical/physical with leaf/unary/binary plan, about 6 plans, no concrete plan e.g. LogicalAggregatePlan). 

But the convenient make negative effect that difficult to understand, and people must known the concept then can develop some new rules, and rule become ugly. So we combine the plan & operator, make the rule as simple as possible, the negative effect is we must overwrite some withXxx for all concrete plan, e.g. LogicalAggregate, PhysicalHashJoin.

## Checklist(Required)

1. Does it affect the original behavior: No
3. Has unit tests been added: No Need
4. Has document been added or modified: No Need
5. Does it need to update dependencies: No
6. Are there any changes that cannot be rolled back: No

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/DateLiteral.java [hunk 1]
```diff
@@ -158,7 +158,8 @@ private enum  DateLiteralType {
         DATEV2(3);
 
         private final int value;
-        private DateLiteralType(int value) {
+
+        DateLiteralType(int value) {
```
[the remark this answers is not in the document]
Comment (924060929) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/DateLiteral.java:162]: fix checkstyle. and private enum DateLiteralType's constructor is also private, so can be omitted

File: fe/fe-core/src/main/java/org/apache/doris/nereids/analyzer/UnboundRelation.java [hunk 2]
```diff
@@ -72,6 +85,16 @@ public LogicalProperties computeLogicalProperties(Plan... inputs) {
...
     }
 
+    @Override
+    public Plan withGroupExpression(Optional<GroupExpression> groupExpression) {
+        return new UnboundRelation(nameParts, groupExpression, Optional.of(logicalProperties));
+    }
+
+    @Override
+    public Plan withLogicalProperties(Optional<LogicalProperties> logicalProperties) {
+        return new UnboundRelation(nameParts, Optional.empty(), logicalProperties);
```
[the remark this answers is not in the document]
Comment (924060929) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/analyzer/UnboundRelation.java:95]: when the plan is changed, the groupExpression must be clear to tell the memo: I create a new plan, don't skip copyIn.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/GroupExpression.java [hunk 3]
```diff
@@ -89,8 +89,8 @@ public void setParent(Group parent) {
         this.parent = parent;
     }
 
-    public Operator getOperator() {
-        return op;
+    public Plan getPlan() {
+        return plan;
```
[the remark this answers is not in the document]
Comment (924060929) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/memo/GroupExpression.java:93]: when the plan copyIn to Memo, the plan will be withLogicalProperties and withChildren(GroupPlan... childrens):

```java
private Plan replaceChildrenToGroupPlan(Plan plan, List<Group> childrenGroups) {
        List<Plan> groupPlanChildren = childrenGroups.stream()
                .map(group -> new GroupPlan(group))
                .collect(ImmutableList.toImmutableList());
        LogicalProperties logicalProperties = plan.getLogicalProperties();
        return plan.withChildren(groupPlanChildren)
            .withLogicalProperties(Optional.of(logicalProperties));
}
```

so the plan in the memo contains the logicalProperties same as the logicalProperties in the groupExpression, and replace GroupPlan as children

**Comment by morningman** [comment]
This PR didn't pass the p0 test, because there are some other issue not related to this PR.
So I skip p0 for this PR.

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**